### ISSUE TITLE
Serial via USB

### ISSUE THREAD

**Issue by toogooda** [Serasidis/STM32_HID_Bootloader#25]
Got it to upload but then com port disappears  is there supposed to be serial support when using this boot-loader?

**Comment by ssieb** [comment]
It uses the HID interface, not serial.

**Comment by toogooda** [comment]
> It uses the HID interface, not serial.

Yes thanks, my question is does that then mean I can't get my serial debug output via the usb because it is being used as a HID or is this part of the bug that then stops all other uploads after the first one?

**Comment by ssieb** [comment]
If you use this bootloader, then you need to use the HID uploader.  Once your sketch is running, it will enable the serial port.
(This is my recollection, I haven't actually used this device for a while.)

**Comment by toogooda** [comment]
> If you use this bootloader, then you need to use the HID uploader. Once your sketch is running, it will enable the serial port.
> (This is my recollection, I haven't actually used this device for a while.)

Great thank you that is what I wanted to know. I am using the ST uploader HID option so there seems to be a bug with it. As it does the upload but then the COM port fails and there is no serial output.

**Comment by toogooda** [comment]
Is this still active? great idea but looks like its not actively being worked on.

**Comment by uzi18** [comment]
what core did you use?
it work so what you want to add/change?

**Comment by toogooda** [comment]
> what core did you use?
> it work so what you want to add/change?
Upload worked but then no serial is possible which is a show stopper.
Tried it in multiple devices.

**Comment by uzi18** [comment]
ok but what core you use for it?

**Comment by toogooda** [comment]
> ok but what core you use for it?

Offical ST one think out is called STM32duino

**Comment by uzi18** [comment]
version?

**Comment by toogooda** [comment]
AT work so not sure sorry will update tonight. Its 9am here :)

**Comment by uzi18** [comment]
10pm here and afterwork :)
